src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,333 @@
+        }
+
+        public static ReadOnlySpan<char> TrimSpaces(ReadOnlySpan<char> span)
+        {
+            if (span.IsEmpty)
+                return span;
+
+            int ichLim = span.Length;
+            int ichMin = 0;
+            if (span[ichMin] != ' ' && span[ichLim - 1] != ' ')
+                return span;
+
+            while (ichMin < ichLim && span[ichMin] == ' ')
+                ichMin++;
+            while (ichMin < ichLim && span[ichLim - 1] == ' ')
+                ichLim--;
+            return span.Slice(ichMin, ichLim - ichMin);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with leading and trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimWhiteSpace(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichMin = 0;
+            int ichLim = memory.Length;
+            var span = memory.Span;
+            if (!char.IsWhiteSpace(span[ichMin]) && !char.IsWhiteSpace(span[ichLim - 1]))
+                return memory;
+
+            while (ichMin < ichLim && char.IsWhiteSpace(span[ichMin]))
+                ichMin++;
+            while (ichMin < ichLim && char.IsWhiteSpace(span[ichLim - 1]))
+                ichLim--;
+
+            return memory.Slice(ichMin, ichLim - ichMin);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="ReadOnlyMemory{T}"/> of <see cref="char"/> with trailing whitespace trimmed.
+        /// </summary>
+        public static ReadOnlyMemory<char> TrimEndWhiteSpace(ReadOnlyMemory<char> memory)
+        {
+            if (memory.IsEmpty)
+                return memory;
+
+            int ichLim = memory.Length;
+            var span = memory.Span;
+            if (!char.IsWhiteSpace(span[ichLim - 1]))
+                return memory;
+
+            while (0 < ichLim && char.IsWhiteSpace(span[ichLim - 1]))
+                ichLim--;
+
+            return memory.Slice(0, ichLim);
+        }
+
+        /// <summary>
+        /// This produces zero for an empty string.
+        /// </summary>
+        public static bool TryParse(ReadOnlyMemory<char> memory, out Single value)
+        {
+            var res = DoubleParser.Parse(out value, memory.Span);
+            Contracts.Assert(res != DoubleParser.Result.Empty || value == 0);
+            return res <= DoubleParser.Result.Empty;
+        }
+
+        /// <summary>
+        /// This produces zero for an empty string.
+        /// </summary>
+        public static bool TryParse(ReadOnlyMemory<char> memory, out Double value)
+        {
+            var res = DoubleParser.Parse(out value, memory.Span);
+            Contracts.Assert(res != DoubleParser.Result.Empty || value == 0);
+            return res <= DoubleParser.Result.Empty;
+        }
+
+        public static uint Hash(ReadOnlySpan<char> span, uint seed) => Hashing.MurmurHash(seed, span);
```

Comment:
A few questions about this function.

1. Why have it when all it does is a simple call forward to another function?  Why not just change all the callers to call the `Hashing` function? That would be one less public method we would need to maintain.
2. If we are going to keep this function, why are the arguments flipped from the MurmurHash function?